### PR TITLE
py-pysal: update with new "family"-ports

### PR DESCRIPTION
**Pull request by nilason**
#### Description

The PySAL (`py-pysal`) package has been refactored, it is now a wrapper package for a number of separate modules.

This PR updates py-pysal and adds the separate modules as new ports.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.6.6 21G646 arm64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by nilason** [comment]
Depends on #19719

**Comment by nilason** [comment]
Oh dear, finally the CI builds succeeded! Please review!

File: python/py-access/Portfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    depends_build-append \
+                    port:py${python.version}-setuptools
+
+    depends_run-append \
+                    port:py${python.version}-requests \
+                    port:py${python.version}-geopandas \
+                    port:py${python.version}-numpy \
+                    port:py${python.version}-pandas
+
+    livecheck.type  none
```
Comment (reneeotten) [inline python/py-access/Portfile:41]: this is the default set by the `python` PG and can thus be removed

File: python/py-access/Portfile [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           python 1.0
+
+github.setup        pysal access 1.1.9 v
```
Comment (reneeotten) [inline python/py-access/Portfile:7]: we typically prefer to fetch Python ports from PyPI - is there a reason as to why you not do that here?

File: python/py-momepy/Portfile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+homepage            http://docs.momepy.org/en/stable/
+
+checksums           rmd160  b4702f95870fbfd48bd4c54f621528fa517ae979 \
+                    sha256  b52cb4e957c0407bd28d7ef260da0288a0c1666e206726eafd12d5976156d42b \
+                    size    21673690
+
+python.versions     39 310 311
+python.pep517       yes
+python.pep517_backend setuptools
```
Comment (reneeotten) [inline python/py-momepy/Portfile:29]: this is the default and shouldn't be specified

File: python/py-pysal/Portfile [hunk 4]
```diff
@@ -4,28 +4,52 @@ PortSystem              1.0
 PortGroup               python 1.0
 PortGroup               github 1.0
 
-github.setup            pysal pysal 1.14.4 v
+github.setup            pysal pysal 23.07 v
 revision                0
-name                    py-pysal
+name                    py-${github.project}
+github.tarball_from     tarball
```
Comment (reneeotten) [inline python/py-pysal/Portfile:10]: why not `archive`? If it needs to be `tarball` then you can omit it, as that is the current default in the `github` PG

File: python/py-spopt/Portfile [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                    port:py${python.version}-scipy \
+                    port:py${python.version}-shapely \
+                    port:py${python.version}-spaghetti \
+                    port:py${python.version}-tqdm
+
+    test.run        yes
+    test.cmd        py.test-${python.branch}
+    test.target
+    test.env-append \
+                    PYTHONPATH=${worksrcpath}/build/lib
```
Comment (reneeotten) [inline python/py-spopt/Portfile:67]: these are the default set up by the `python` PG ; so you'll only need to retain `test.run yes`

File: python/py-spopt/Portfile [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+python.versions     39 310 311
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+                    port:py${python.version}-setuptools \
+                    port:py${python.version}-setuptools_scm
+
+    depends_test-append \
+                    port:py${python.version}-pytest \
```
Comment (reneeotten) [inline python/py-spopt/Portfile:40]: this is added by default when setting `test.run yes` so the `pytest` dependency can be removed here

File: python/py-spopt/Portfile [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+                    port:py${python.version}-spaghetti \
+                    port:py${python.version}-tqdm
+
+    test.run        yes
+    test.cmd        py.test-${python.branch}
+    test.target
+    test.env-append \
+                    PYTHONPATH=${worksrcpath}/build/lib
+
+    livecheck.type  none
```
Comment (reneeotten) [inline python/py-spopt/Portfile:69]: default

**Review by reneeotten — CHANGES_REQUESTED** [review]
thanks @nilason for all your work on these ports! I've left a few comments - they all apply to pretty much every port in this PR.

**Comment by nilason** [comment]
> thanks @nilason for all your work on these ports! I've left a few comments - they all apply to pretty much every port in this PR.

I believe I have addressed all your points. I kept one port (py-spopt) installing from Github and therefore enabling the tests.

(Some other modules also did have tests, but they seemed unreliable.)

**Review by reneeotten — APPROVED** [review]

**Comment by reneeotten** [comment]
thanks @nilason !

**Comment by nilason** [comment]
@reneeotten Thank you!

**Pull request merged**